src/mtype.d

```diff
@@ -9497,27 +9453,69 @@ public:
         return params;
     }
 
+    extern (D):
```

Comment:
It won't compile as extern(C++). Refactoring to remove all the unnecessary linkage specifications is way beyond the scope of this PR.